BHoM_UI/Components/Engine/External.cs

```diff
@@ -20,52 +20,43 @@
  * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
  */
 
-using BH.Engine.Reflection;
-using BH.oM.UI;
+using BH.UI.Templates;
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
+using System.Reflection;
+using BH.Adapter;
 
-namespace BH.Engine.UI
+namespace BH.UI.Components
 {
-    public static partial class Query
+    public class ExternalCaller : MethodCaller
     {
-        /***************************************************/
-        /**** Public Methods - Interfaces               ****/
-        /***************************************************/
+        /*************************************/
+        /**** Properties                  ****/
+        /*************************************/
 
-        public static bool IIsToKeepInMenu(this object obj)
-        {
-            return IsToKeepInMenu(obj as dynamic);
-        }
+        public override System.Drawing.Bitmap Icon_24x24 { get; protected set; } = Properties.Resources.External;
 
+        public override Guid Id { get; protected set; } = new Guid("4DEC2B78-4A83-49A2-BD7D-A03DD5CAE43E");
 
-        /***************************************************/
-        /**** Public Methods                            ****/
-        /***************************************************/
+        public override string Name { get; protected set; } = "ExternalCompute";
```

Comment:
I would prefer it to be name `External` instead of `ExternalCompute`. 
I understand where you're going with this but not every linked method is going to correspond to a `Compute` format so I'd rather stick to `External` to avoid the parallel.